src/bin/pgcopydb/schema.c

```diff
@@ -971,14 +971,19 @@ struct FilteringQueries listSourceTablesSQL[] = {
 		"               ("
 		"                  select attnum, atttypid::integer, "
 		"                         format('%I', attname) as attname, "
-		"                         i.indrelid is not null as attisprimary "
+		"                         i.indrelid is not null as attisprimary, "
+		"						  col.is_generated = 'ALWAYS' as attisgenerated "
 		"                    from pg_attribute a "
 		"                         left join pg_index i "
-		"                                on i.indrelid = a.attrelid "
+		"                               on i.indrelid = a.attrelid "
 		"                               and a.attnum = ANY(i.indkey) "
 		"                               and i.indisprimary "
+		"					 	  left join information_schema.columns col "
+        "                    			on col.column_name = a.attname "
 		"                   where a.attrelid = c.oid and not a.attisdropped "
 		"                     and a.attnum > 0 "
+		"					  and col.table_name = c.relname "
+		"          			  and col.table_schema = n.nspname "		
```

Comment:
@dimitri, these conditions ensure we have the right column

Reply:
Well then they need to be part of the JOIN condition (modern SQL style, from 1989 onwards I believe) rather than in the WHERE clause (legacy SQL style).